refactor(ast_codegen): remove `Generator::name` and `Pass::name` methods

overlookmotel opened the pull request:
Remove `Generator::name` and `Pass::name` methods. All impls for these methods return a string identical to the struct name, so can set return value of `Runner::name` in `define_generator!` and `define_pass!` macros instead.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4764?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4764** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4764?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#4763** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4763?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#4762** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4762?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#4761** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4761?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

rzvxa reviewed (commented):
I use this method to allow custom identifiers for generators/passes.
I leverage this for generating debug information.

```rust
.pass(AssertX("Test1"))
.pass(YYY)
.pass(AssertX("Test2"))
```

At the moment it doesn't have any use upstream but is useful for local development.
However, I can implement those manually. It doesn't have to go through the macro.

overlookmotel committed:
refactor(ast_codegen): remove `Generator::name` and `Pass::name` methods (#4764)

Remove `Generator::name` and `Pass::name` methods. All impls for these methods return a string identical to the struct name, so can set return value of `Runner::name` in `define_generator!` and `define_pass!` macros instead.

overlookmotel commented:
> I use this method to allow custom identifiers for generators/passes. I leverage this for generating debug information.

I can't say I understand you. But feel free to revert this if those methods are actually useful. I just couldn't see them being used anywhere at present, but I may well be missing some context.
